client/my-sites/domains/domain-management/settings/cards/domain-forwarding-card.tsx

```diff
@@ -110,14 +111,29 @@ export default function DomainForwardingCard( { domain }: { domain: ResponseDoma
 			const url = new URL( forwarding.targetPath, origin );
 			if ( url.hostname !== '_invalid_.domain' ) {
 				setTargetUrl( url.hostname + url.pathname + url.search + url.hash );
-				setProtocol( forwarding.isSecure ? 'https' : 'http' );
 				setIsPermanent( forwarding.isPermanent );
 				setForwardPaths( forwarding.forwardPaths );
+				setSubdomain( forwarding.subdomain );
+				setSourceType( forwarding.subdomain !== '' ? 'subdomain' : 'domain' );
 			}
 		} catch ( e ) {
 			// ignore
 		}
-	}, [ isLoading, forwarding, setTargetUrl, setProtocol ] );
+	}, [ isLoading, forwarding, setTargetUrl ] );
+
+	const handleDomainOriginChange = ( event: React.ChangeEvent< HTMLInputElement > ) => {
+		setSubdomain( withoutHttp( event.target.value ) );
```

Comment:
Wondering if it helps with readability by setting `event.target.value` to a variable as it is used 3 times in this function.

Reply:
Thanks for the suggestions @candy02058912. [Fixed them here](https://github.com/Automattic/wp-calypso/pull/81047/commits/e77388ace9861d71fe3c2e27b14cc269aadf13c4).

Reply:
> We also need to add this constraint to the back end.

Follow-up task added here: https://github.com/Automattic/wp-calypso/pull/81047#discussion_r1312987017